controller/api/destination/endpoint_profile_translator.go

```diff
@@ -1,69 +1,136 @@
 package destination
 
 import (
-	"fmt"
-
 	pb "github.com/linkerd/linkerd2-proxy-api/go/destination"
 	"github.com/linkerd/linkerd2/controller/api/destination/watcher"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	logging "github.com/sirupsen/logrus"
+	"google.golang.org/protobuf/proto"
 )
 
 type endpointProfileTranslator struct {
 	enableH2Upgrade     bool
 	controllerNS        string
 	identityTrustDomain string
 	defaultOpaquePorts  map[uint32]struct{}
-	stream              pb.Destination_GetProfileServer
-	lastMessage         string
+
+	stream    pb.Destination_GetProfileServer
+	endStream chan struct{}
+
+	updates chan *watcher.Address
+	stop    chan struct{}
+
+	current *pb.DestinationProfile
 
 	log *logging.Entry
 }
 
-// newEndpointProfileTranslator translates pod updates and protocol updates to
+// endpointProfileUpdatesQueueOverflowCounter is a prometheus counter that is incremented
+// whenever the profile updates queue overflows.
+//
+// We omit ip and port labels because they are high cardinality.
+var endpointProfileUpdatesQueueOverflowCounter = promauto.NewCounter(
+	prometheus.CounterOpts{
+		Name: "endpoint_profile_updates_queue_overflow",
+		Help: "A counter incremented whenever the endpoint profile updates queue overflows",
+	},
+)
```

Comment:
No label for service?

Reply:
These lookups are specifically for non-service IPs.

Reply:
It's also used when getting profile by name, so yeah that label won't make sense in all cases.